Go/search/interpolation_search.go

```diff
@@ -0,0 +1,45 @@
+package main
+import "fmt"
+ 
+func Search(array []int, number int) int {
+	
+	low := 0
+	high := len(array)-1
+ 
+    for {
+	var key int
+	var size int
+	
+        if high == low {
+            key = high
+        
+	} else {
+		size = high - low
+		key = low + (int(float64(size-1)*(float64(number-array[0])/float64(array[len(array)-1]-array[0]))))
```

Comment:
Can you make it more compact and readable by breaking down the statement into multiple other lines?